src/Reveal.Sdk.Dom.Tests/Core/Utilities/RdashDocumentValidatorFixture.cs

```diff
@@ -131,21 +131,46 @@ public void Validate_PreventsDuplicate_Fields()
         }
 
         [Fact]
-        public void Validate_Adds_JoinTables()
+        public void Validate_PreventsDuplicate_JoinTables()
         {
             var dataSourceItem = new DataSourceItem("Test", new DataSource()).SetFields(new List<IField> { new TextField() });
             var joinConditions = new List<JoinCondition> { new JoinCondition("left", "right") };
             var dataSourceItemToJoin = new DataSourceItem().SetFields(new List<IField> { new TextField() });
 
+            dataSourceItem.Join("Alias", joinConditions, dataSourceItemToJoin);
             dataSourceItem.Join("Alias", joinConditions, dataSourceItemToJoin);
 
             var document = new RdashDocument();
             document.Visualizations.Add(new GridVisualization(dataSourceItem));
 
-            Assert.Empty(document.Visualizations[0].DataDefinition.AsTabular().JoinTables);
+            Assert.Equal(2,document.Visualizations[0].DataDefinition.AsTabular().JoinTables.Count);
+
+            RdashDocumentValidator.Validate(document);
+
+            Assert.Single(document.Visualizations[0].DataDefinition.AsTabular().JoinTables);
+        }
+
+        [Fact]
+        public void Validate_JoinTables_JoinsAreNotDuplicated()

```

Comment:
This test name is the same as the test above. What is this testing that is different?

Also, we need a test to cover the `Join(string alias, string leftJoinFieldName, string rightJoinFieldName, DataSourceItem dataSourceItem)` method too. I would have one test use one of the method signatures, and another test for the other

Reply:
The first test is covering that the validate call removes duplicates (I'm including the same alias twice), while the second one is just checking that multiple calls to validate won't add the same alias again. Renamed the second test to follow the naming convention. Added a new test for the second Join signature.